File: EmployeeMonitoringClient/src/app/model/task.ts
export enum TaskStatus {
    UNASSIGNED = 'UNASSIGNED',
    ASSIGNED = 'ASSIGNED',
    COMPLETED = 'COMPLETED'
}

export interface Task {
    id?: number;
    name: string;
    description: string;
    deadline: string;
    status: TaskStatus;
}

File: EmployeeMonitoring/src/main/java/empl/monitor/EmployeeMonitoring/model/Task.java
package empl.monitor.EmployeeMonitoring.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;

    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime deadline;

    @Enumerated(EnumType.STRING)
    private TaskStatus status;
}
